docs/networking/streamcodecs.md

```diff
@@ -172,20 +176,22 @@ public static final StreamCodec<RegistryFriendlyByteBuf, Integer> STREAM_CODEC =
 
 A stream codec which supplies an in-code value and encodes to nothing can be represented using `StreamCodec#unit`. This is useful if no information should be synced across the network.
 
+:::warning
+Unit stream codecs expects that any encoded object must match the unit specified; otherwise an error will be thrown. Therefore, all objects must have some `equals` implementation that returns true for the unit object.
```

Comment:
```suggestion
Unit stream codecs expect that any encoded object must match the unit specified; otherwise an error will be thrown. Therefore, all objects must have some `equals` implementation that returns true for the unit object.
```
That's not quite true. Using objects with default equality behaviour (i.e. reference equality) is fine, just need to make sure to use the specific instance in every case.

Reply:
Yes, though my general thought process is that if you're supplying something within a stream codec, it is unlikely going to be the same specific instance; though I could just mention it still.